test/test-signal.js

```diff
@@ -0,0 +1,64 @@
+'use strict';
+
+// Testcase to produce NodeReport on signal
+if (process.argv[2] === 'child') {
+  require('../').setEvents("signal");
+
+  // Exit on loss of parent process
+  process.on('disconnect', () => process.exit(2));
+
+  function myLoop() {
+    var list = [];
+    for (var i=0; i<10000000000; i++) {
+      for (var j=0; i<1000; i++) {
+        list.push(new MyRecord());
+      }
+      for (var j=0; i<1000; i++) {
+        list[j].id += 1;
+        list[j].account += 2;
+      }
+      for (var j=0; i<1000; i++) {
+        list.pop();
+      }
+    }
+  }
+
+  function MyRecord() {
+    this.name = 'foo';
+    this.id = 128;
+    this.account = 98454324;
+  }
+
+  process.send('child started');
+  myLoop();
+} else {
+  const common = require('./common.js');
+  const fork = require('child_process').fork;
+  const fs = require('fs');
+  const tap = require('tap');
+
+  if (process.platform === 'win32') {
+    tap.fail('Unsupported on Windows', { skip: true });
+    return;
+  }
+
+  const child = fork(__filename, ['child'], { 'silent': true });
+  // Wait for child to signal it is ready before sending signal
+  child.on('message', () => {
+    // Wait a bit before sending signal to allow the child to enter myLoop()
```

Comment:
Okay, we do need a delay otherwise the process exits before writing the report. I have a better idea -- Scan `stderr` from the child for `Node.js report completed`, which is output after the report has been written, and send the `SIGTERM` when the message is observed.